### ISSUE TITLE
Recent Files does not work: Stays empty

### ISSUE THREAD

**Issue by teamcons** [elementary/files#2497]
### What Happened?

Nothing shows up in Recent Files in the file manager

Zeitgeist is running, and the privacy tabs follows default

Disabling then enable history does not change anything

### Steps to Reproduce

Default install, nothing fancy done. Same situation in the last few weeks

### Expected Behavior

Recent Files shows up in Recent Files

### OS Version

8.x (Early Access)

### Session Type

Secure Session (Wayland)

### Software Version

Latest release (I have run all updates)

**Comment by jeremypw** [comment]
Thanks for the report. Please confirm that the view shows "There are no Recent Files" rather than "Privacy Mode On".  If you disable history and re-enable then recent files are cleared so you would need to add some more. Not all apps use the "recent://" location for this.  GEdit definitely does.  (See https://github.com/elementary/code/pull/1461 for a PR to make Code do the same).  Note also https://github.com/elementary/files/issues/2496 - the recent view does not always automatically update after switching history off and on, so needs to be refreshed.

**Comment by teamcons** [comment]
I confirm it shows "there are no recent files"

Actually it showed at first, post install only one file: The profile picture of my account, which on first boot was the only file in $HOME
Later on, no subsequent file showed up

I have tried to unstick "Recent Files" by activating and deactivating privacy mode (and clearing history? im not sure) a few reboots ago, which only cleared it from that only file

i never file any bug report without witnessing a bug accross a few different sessions (so with reboots inbetween)

**Comment by jeremypw** [comment]
I just tried editing a file using the *Flatpak* version of GEdit and it did *not* add it to `recent://` where as the apt installed version did.   From which apps are you expecting to see recent files and how were they installed?

You could try installing `nautilus` (using `sudo apt install`) and seeing whether that shows  any files in `recent://`.  I have been unable to reproduce this bug on OS8 when there are definitely recent files in `recent://`.

**Comment by teamcons** [comment]
> From which apps are you expecting to see recent files and how were they installed?

Usage is editing textfiles with Code,
Looking at pictures with Photos
taking screenshots with the eOS default screenshot thingie
opening various images with discord
editing excel and word files with OnlyOffice
Listening to music files with Music
watching videos with Videos, then Celluloid (Videos not functional here)
Opening PDFs with the default PDF viewer
saving a file with Mail

all Flatpaks.

I did today a fresh reinstall, the only image showing up is the profile picture of another user account i set up (that crashes, i need to file a bug report)

> Nautilus

Will try, will keep you updated

**Comment by jeremypw** [comment]
Yes, I think `zeitgeist` doesn't work with Flatpaks or is being abandoned in favour of Tracker. Maybe there is a portal for adding files to a global recent file manager (I'll check) but I Flatpak apps maintain their own list within the sandbox?  I am pretty sure this isn't a Files bug as such - it shows recent files added to `recent://` but a lot of apps (including Files) are not adding them as you expect.

**Comment by jeremypw** [comment]
I notice I raised related issues #2046 and #2274 last year. And there are some even older ones #1671 and #1181.  As most apps are likely to run in a sandbox in future the solution lies outside of Files I think.
